addons/calendar/tests/test_event_notifications.py

```diff
@@ -44,11 +45,29 @@ def test_message_invite_allday(self):
                 'partner_ids': [(4, self.partner.id)],
             }])
 
-
     def test_message_invite_self(self):
+        # The organizer is used as the author
+        self.event.user_id = self.user
+        with self.assertNoNotifications():
+            self.event.with_user(self.user_root).partner_ids = self.partner
+        # Fallback on current user if no organizer
+        self.event.user_id = False
         with self.assertNoNotifications():
             self.event.with_user(self.user).partner_ids = self.partner
 
+    def test_message_invite_self_notify(self):
+        # The organizer is used as the author
+        self.event.user_id = self.user
+        with self.assertSinglePostNotifications([{'partner': self.partner, 'type': 'inbox'}], {
+            'message_type': 'user_notification',
+            'subtype': 'mail.mt_note',
+        }):
+            self.event.with_user(self.user_root).with_context(mail_notify_author=True).partner_ids = self.partner
```

Comment:
Same thing here, it's confusing as "mail_notify_author" is not used to notify the "author of the change" (user_root) but rather the "author aka the organizer", does not really feel right.